src/ApplicationLogs/Store/States/BlockLogState.cs

```diff
@@ -16,9 +16,9 @@ namespace ApplicationLogs.Store.States
 {
     public class BlockLogState : ISerializable, IEquatable<BlockLogState>
     {
-        public Guid[] NotifyLogIds { get; private set; } = [];
+        public Guid[] NotifyLogIds { get; private set; }
```

Comment:
We should make this `required` as well.

Reply:
like this? Kinda ugly:

```C#
public virtual void Deserialize(ref MemoryReader reader)
{
    // It should be safe because it filled from a block's notifications.
    var aLen = reader.ReadUInt32();
    var notifyLogIds = new Guid[aLen];
    for (var i = 0; i < aLen; i++)
        notifyLogIds[i] = new Guid(reader.ReadVarMemory().Span);

    // Create a new instance of BlockLogState using the Create factory method
    var blockLogState = Create(notifyLogIds);

    // Assign the created instance to the current object
    NotifyLogIds = blockLogState.NotifyLogIds;
}

```